includes/analytics/class-analytics.php

```diff
@@ -42,18 +44,25 @@ public function handle_event( string $name, array $context ) {
 			return;
 		}
 
-		wp_remote_post(
-			$this->event_url,
+		/**
+		 * Filters the URL used to send events to.
+		 *
+		 * @param string The URL to use when sending events.
+		 */
+		$url = apply_filters( 'stellarwp/analytics/event_url', self::DOMAIN . self::ENDPOINT );
+
+		$response = wp_remote_post(
+			$url,
 			array(
 				'timeout' => 20,
```

Comment:
`'data_format' => 'body',` is required to properly send JSON with WordPress's client.

```suggestion
				'timeout' => 20,
				'data_format' => 'body',
```

Reply:
[It defaults to that](https://github.com/WordPress/WordPress/blob/832dd52aded67dd2830e2f73d869c788e5140539/wp-includes/class-wp-http.php#L363C3-L366C4) for POST requests already.

Reply:
TIL!